### ISSUE TITLE
Select segment ID (Annonation) throws an error

### ISSUE THREAD

**Issue by vzinche** [mobie/mobie-viewer-fiji#911]
![Screenshot from 2022-11-25 13-54-58](https://user-images.githubusercontent.com/32451366/203990356-5aa3cbaf-16b3-457c-b22e-1538f5c88245.png)

**Comment by tischi** [comment]
@constantinpape @vzinche 

Probably we had this discussion, but to be sure....:

It is not so clear what the ID should be.

For segments, spots, and regions we have an UUID, which is what you would use in the JSON.

- Segments & Spots: `String uuid = source + ";" + timePoint + ";" + labelId;`
- Regions: `String uuid = timePoint + ";" + regionId;` (note that regionId is a String)

We want the annotator to work for segments, spots, and regions...

Upon instantiation, the Annotator code can know which type it annotates.

Currently the Annotator matches the UUID, but I admit this is too much to type in practice.

I am not sure now what to do... 

For example, I guess for the covid-if plate just `labelId` for a `Segment` is not enough, is it?

**Comment by tischi** [comment]
One idea: 

If it is a spot or segment; I could try to parse the user entry to an integer; if that works, I would look for a matching `label_id`. If that returns multiple matches I would throw an error and tell the user to please be more specific, i.e. use the full UUID.

What do you think?

**Comment by vzinche** [comment]
To me this sounds good! But I am only annotating labelIds, so my opinion is biased :)

**Comment by tischi** [comment]
@constantinpape 

I am not sure, but I think what I did should work as well?

```
		final String annotationId = selectAnnotationTextField.getText();

		List< A > selectedAnnotations;
		try
		{
			// For spots and segments the entered ID may be just the
			// label id; as this is a frequently occurring use-case, try this first.
			final int labelId = Integer.parseInt( annotationId );
			selectedAnnotations = annotations.parallelStream().filter( a -> a.label() == labelId ).collect( Collectors.toList() );
		}
		catch ( Exception e )
		{
			// Match the full UUID
			selectedAnnotations = annotations.parallelStream().filter( a -> a.uuid().equals( annotationId ) ).collect( Collectors.toList() );
		}

		if ( selectedAnnotations.size() == 0 )
			throw new RuntimeException( "Could not find " + annotationId );

		if ( selectedAnnotations.size() > 1 )
			throw new RuntimeException( "Found multiple matches of " + annotationId );

		return selectedAnnotations.get( 0 );
```

**Comment by constantinpape** [comment]
> It does work as far as I could test, but I am not sure I am missing some (edge) case?

I am pretty sure this is missing the case where a user passes a regionId without the timepoint. It will fail the first `try` block, because the region id is (usually) not an integer. And it also doesn't match the `catch` block because the user hasn't passed the full UUID.

**Comment by tischi** [comment]
Should be mainly fixed with `3.0.7`. 
I would like to close this issue.
Opened a new one for future improvements: https://github.com/mobie/mobie-viewer-fiji/issues/913
